bitbucket-endpoint/src/main/java/com/microfocus/adm/almoctane/integration/git/bitbucketendpoint/entity/BitbucketBranch.java

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 EntIT Software LLC, a Micro Focus company, L.P.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package com.microfocus.adm.almoctane.integration.git.bitbucketendpoint.entity;
+
+import com.google.api.client.util.Key;
+import com.microfocus.adm.almoctane.integration.git.bitbucketendpoint.urls.BitbucketBrowseCodeUrl;
+import com.microfocus.adm.almoctane.integration.git.common.entities.Branch;
+
+/**
+ * Class used to describe the branch information received from the Bitbucket response
+ */
+public class BitbucketBranch implements Branch {
+    @Key("id")
+    private String id;
+
+    @Key("displayId")
+    private String branchName;
+
+    private String browseCodeOnBranchUrl;
+
+    private String repositoryName;
+
+    /**
+     * @return - The name of the branch.
+     */
+    @Override
+    public String getBranchId() {
+        return this.id;
+    }
+
+    /**
+     * @return - The full name of the Bitbucket branch.
+     */
+    @Override
+    public String getBranchName() {
+        return this.branchName;
+    }
+
+    /**
+     * Sets the branch browse URL.
+     *
+     * @param server      - The Bitbucket server.
+     * @param projectName - The Bitbucket projectName name.
+     */
+    public void setBrowseCodeOnBranchUrl(String server, String projectName) {
```

Comment:
creating a GenericUrl object which doesn`t do anything else besides setting this field seems a little wasteful and considering BitbucketBrowseCodeUrl doesn`t do anything else i don`t see the point of it.